libs/pika/threading_base/src/thread_num_tss.cpp

```diff
@@ -61,7 +61,7 @@ namespace pika { namespace threads { namespace detail {
         thread_nums_tss() = t;
     }
 
-    thread_nums get_thread_nums_tss()
+    const thread_nums& get_thread_nums_tss()
```

Comment:
How about replacing this with this definition instead: https://github.com/pika-org/pika/blob/79866e4c6458a258a9c1e4535ed87274f0f4f137/libs/pika/threading_base/src/thread_num_tss.cpp#L18-L23? I.e. keep the `get_thread_nums_tss` name, but but the `static thread_local` variable inside this function, and then use `get_thread_nums_tss` instead of `thread_nums_tss` in these spots: https://github.com/pika-org/pika/blob/79866e4c6458a258a9c1e4535ed87274f0f4f137/libs/pika/threading_base/src/thread_num_tss.cpp#L28?

Finally, you can replace these accesses: https://github.com/pika-org/pika/blob/0ab7223c4e08a9888bca38260ccb53db5bd0c2c0/libs/pika/schedulers/include/pika/schedulers/shared_priority_queue_scheduler.hpp#L328-L331 with `get_global_thread_num_tss` etc.

Reply:
I don't really know what you're trying to say.

Reply:
With your current change `get_thread_nums_tss` has exactly the same signature as `thread_nums_tss` and all `get_thread_nums_tss` does is call `thread_nums_tss`. So I'm suggesting to remove the unnecessary function call.

The second suggestion is that you would also avoid the potential copy by using the helper functions `get_global_thread_num_tss` etc.